source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
My understanding is that envoy requires explicit port suffixes in hostname strings (e.g. domains in virtual hosts), just like RFC says. 
I'd like to have the port number to be a part of the host rewrite string, and to be consistent with the server side config, it probably should be part of the string literal in the client cluster config. So basically, either strip port on both client and server side, or don't, but be consistent.